src/main/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngine.java

```diff
@@ -59,33 +58,33 @@ public ProbeEngine(List<Probe> probes, PluginService pluginService) {
      */
     public void run() {
         LOGGER.info("Start running probes on all plugins");
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Registered probes: {}", probes.stream().map(Probe::key).collect(Collectors.joining(", ")));
-        }
         pluginService.streamAll().parallel()
-            .peek(plugin -> probes.stream()
-                .map(probe -> {
+            .peek(plugin -> {
+                probes.forEach(probe -> {
```

Comment:
In main branch, currently, we're using `probes.stream().forEach()` and here you've replaced that with`probes.forEach()`.

The main aim of this PR is to pass the context of 1 probe to another probe in a pre-determined sequential way. And as per the check [done by me here](https://github.com/jenkins-infra/plugin-health-scoring/blob/172e52b3ebd8bc26301185f02f21cc1bfc59d4f4/src/main/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbe.java#L39), this won't work because it is using `probes.stream().forEach()` therefore there is a possibility that my new probe would run "before" the `SCMLinkValidationProbe`. I have questions:

1. While using `probes.stream().forEach()`, would our probes be running as per the `ORDER` defined by us in each probe file?
2. While using `probes.forEach()`, the probes would be running based on the iterator which uses the value of `ORDER` variable to order the execution of each probe. Is this correct?

Reply:
there is no guaranty of order of execution with the Stream.
In our case, because we might need a previous probe result to execution the current probe, we need to be sure of the execution order. So, we need to be sure that right after the probe is execution, that we have a result, we put this result into the plugin details.